pints/_abc/__init__.py

```diff
@@ -0,0 +1,366 @@
+            raise ValueError('Number of parameters in prior must match number '
+                             'of parameters in model')
+
+        # Get number of parameters
+        self._n_parameters = self._log_prior.n_parameters()
+
+        # Set default method
+        if method is None:
+            method = pints.RejectionABC
+        else:
+            try:
+                ok = issubclass(method, pints.ABCSampler)
+            except TypeError:   # Not a class
+                ok = False
+            if not ok:
+                raise ValueError('Given method must extend pints.ABCSampler.')
+
+        # Create sampler
+        self._sampler = method(log_prior)
+
+        # Target number of samples
+        self._n_target = 500
+        self.set_n_target()
+
+        # Logging
+        self._log_to_screen = True
+        self._log_filename = None
+        self._log_csv = False
+        self.set_log_interval()
+
+        # Parallelisation
+        self._parallel = False
+        self._n_workers = 1
+        self.set_parallel()
+
+        #
+        # Stopping criteria
+        #
+
+        # Maximum iterations
+        self._max_iterations = 10000
+
+
+    def set_log_interval(self, iters=20, warm_up=3):
+        """
+        Changes the frequency with which messages are logged.
+
+        Paramaters
+        ----------
+        interval
+            A log message will be shown every ``iters`` iterations.
+        warm_up
+            A log message will be shown every iteration, for the first
+            ``warm_up`` iterations.
+
+        """
+        iters = int(iters)
+        if iters < 1:
+            raise ValueError('Interval must be greater than zero.')
+        warm_up = max(0, int(warm_up))
+
+        self._message_interval = iters
+        self._message_warm_up = warm_up
+
+    def set_log_to_file(self, filename=None, csv=False):
+        """
+        Enables progress logging to file when a filename is passed in, disables
+        it if ``filename`` is ``False`` or ``None``.
+
+        The argument ``csv`` can be set to ``True`` to write the file in comma
+        separated value (CSV) format. By default, the file contents will be
+        similar to the output on screen.
+        """
+        if filename:
+            self._log_filename = str(filename)
+            self._log_csv = True if csv else False
+        else:
+            self._log_filename = None
+            self._log_csv = False
+
+    def set_log_to_screen(self, enabled):
+        """
+        Enables or disables progress logging to screen.
+        """
+        self._log_to_screen = True if enabled else False
+
+    def max_iterations(self):
+        """
+        Returns the maximum iterations if this stopping criterion is set, or
+        ``None`` if it is not. See :meth:`set_max_iterations()`.
+        """
+        return self._max_iterations
+
+    def n_target(self):
+        """
+        Returns the target number of samples to obtain in the estimated
```

Comment:
When you say "target", is this guaranteed to be met? If so, maybe leave out the word target in both the dosctring and the name and just say its the number of samples to sample?
(See also set_n_target)

Reply:
Guessing from the code we might return _more_ than the target?